File: Backend/Domain/Models/Digimon.cs
using Backend.Domain.Models.Digimons;

namespace Backend.Domain.Models
{
    public record class Digimon
    {
        public int Level { get; set; }
        public int Experience { get; set; }
        public Vitals Vitals { get; set; } = new();
        public Attributes Attributes { get; set; } = new();
        public Resistances Resistances { get; set; } = new();
        public Equipments Equipments { get; set; } = new();
        public List<DigievolutionSlot> Digievolutions { get; set; } = [];
        public int ActiveDigievolutionId { get; set; }

        public virtual bool Equals(Digimon? other)
        {
            if (other is null) return false;

            return Vitals.Equals(other.Vitals) &&
                   Attributes.Equals(other.Attributes) &&
                   Resistances.Equals(other.Resistances) &&
                   Equipments.Equals(other.Equipments) &&
                   ActiveDigievolutionId == other.ActiveDigievolutionId &&
                   Digievolutions.SequenceEqual(other.Digievolutions);
        }

        public override int GetHashCode()
        {
            var hash = new HashCode();
            hash.Add(Vitals);
            hash.Add(Attributes);
            hash.Add(Resistances);
            hash.Add(Equipments);
            hash.Add(ActiveDigievolutionId);
            foreach (var d in Digievolutions)
            {
                hash.Add(d);
            }
            return hash.ToHashCode();
        }
    }
}

File: Frontend/src/models/party/digimon/digimon.ts
import type { Vital } from "./vital";
import type { InBattle } from "./in-battle";
import type { Attributes } from "./attributes";
import type { Resistances } from "./resistances";
import type { Equipments } from "./equipments";
import type { DigievolutionSlot } from "./digievolution-slot";
import type { StoredDigievolution } from "./stored-digievolution";

export interface Digimon {
  level: number;
  tp: number;
  blast: number;
  experience: number;
  hp: Vital;
  mp: Vital;
  inBattle: InBattle;
  attributes: Attributes;
  resistances: Resistances;
  equipments: Equipments;
  digievolutions: DigievolutionSlot[];
  storedDigievolutions: StoredDigievolution[];
  activeDigievolutionId: number | null;
}
